block-producer/src/server/api.rs

```diff
@@ -38,9 +40,14 @@ where
     ) -> Result<tonic::Response<SubmitProvenTransactionResponse>, Status> {
         let request = request.into_inner();
 
+        let request_id = gen_request_id();
+        let _span = info_span!("submit_proven_transaction", request_id, ?request, COMPONENT);
```

Comment:
A couple of comments here:

First, I'd probably get rid of `request_id` here as I'm not sure it adds much value right now. My understanding is that we'd want to use request IDs for correlating requests across different components, and this is a bigger issue that we'll address in subsequent PRs.

Second, this would log the entire request at `info` level, right? If so, I think this could be quite verbose as these requests could be 60KB - 100KB of data (STARK proofs are pretty big). Plus, I think the request is mostly binary data. If we do want to log the entire request, we should probably do this at the `trace` level (but would still keep the span at the `info` level).

Also, general question (mostly for my understanding): this span will be dropped at the end of the function, right? So, if an error happens anywhere below, will it be captured in this span?

Reply:
@bobbinth thank you for clarifications. I will remove `request_id`. I agree, it will be better to hide only STARK proofs under `trace`. Spans are not able to catch results by themselves, in this case we would need to process it by hands, thank you for noticing. I will move span to `instrument` since we don't need `request_id` here anymore, `instrument` does support logging of results and errors.